backend/src/main/java/com/eshka/entity/User.java

```diff
@@ -28,6 +28,7 @@
     @Id
     @GeneratedValue(strategy = GenerationType.AUTO)
     private Long userId;
+    private Long teamId;
```

Comment:
Here should be `Team team` instead of `Long teamId`.
Also, don't forget to specify relationship type.

Reply:
```java
@ManyToOne(fetch = FetchType.LAZY, cascade = {CascadeType.MERGE, CascadeType.PERSIST})
@JoinColumn(name = "team_id")
private Team team;
```